nodecontrol/algodControl.go

```diff
@@ -40,6 +40,22 @@ const StdErrFilename = "algod-err.log"
 // StdOutFilename is the name of the file in <datadir> where stdout will be captured if not redirected to host
 const StdOutFilename = "algod-out.log"
 
+type NodeAlreadyStoppedError struct {
+	algodDataDir string
+}
+
+func (e *NodeAlreadyStoppedError) Error() string {
+	return fmt.Sprintf("Node in directory %s has already stopped", e.algodDataDir)
+}
+
+type InvalidDataDirError struct {
+	algodDataDir string
+}
+
+func (e *InvalidDataDirError) Error() string {
+	return fmt.Sprintf("%s is not a valid directory", e.algodDataDir)
```

Comment:
This is not the "correct" error for this use case. using util.IsDir would return true if the directory exists.
It doesn't tell you if a directory of that name could be created.
In that case, you want a "MissingDataDirError" which would format into
"the provided data directory '%s' does not exists"
You could get more information by calling os.Stat and getting the precise underlying error.